### PR TITLE
feat: add `prepr` cms image provider integration

### PR DESCRIPTION
**Pull request by FranciscoKloganB**
# Description

This PR adds an integration with [Prepr Headless CMS](https://prepr.io/), docs can be found at [Prepr Docs](https://docs.prepr.io/).

## Motivation and Context

While the image transformation options provided by this CMS are still few, their Support team informs me, that they are due to receive updates very soon. I also happen to need to use this provider on my personal project and I much prefer to share the provider with the community then have it all for myself. :heart:

## Changes

- Sorts `BuiltInProviders` array alphabetically, no removals were made, only `prepr` key was added.
- Add `src/runtime/prepr` folder containing the necessary code, to `getImage`.
- Add prepr documentation to `docs/content`
- Add necessary configurations for `playground`

### CODE REVIEW

**Comment by danielroe** [comment]
Would you be able to add snapshots/expected values for prepr? 🙏

**Comment by FranciscoKloganB** [comment]
@danielroe Hey, I think anything is possible. Can you just clarity what you are looking for? Thanks

**Comment by danielroe** [comment]
Merge in the latest changes in the main branch, and get the tests passing. (This requires adding `prepr` entries to the unit tests, just like the other providers.)

**Comment by FranciscoKloganB** [comment]
Makes sense. Will do. :)

**Comment by FranciscoKloganB** [comment]
@danielroe I think that's it :)

File: docs/content/5.providers/prepr.md [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+to Prepr Support team, more modifiers will be added soon, and the documentation
+page is due for an update.
+
+For the time being you might find the following links useful:
+
+- [Assets Resizing via REST API](https://docs.prepr.io/reference/rest/v1/assets-resizing/)
+- [Understanding your marketing and design team workflows](https://docs.prepr.io/managing-content/images)
+
+::alert{type="info"}
+Unfortunatly prepr.io still does not provide away to restrict what domains can
```
Comment (danielroe) [inline docs/content/5.providers/prepr.md:34]:
```suggestion
prepr.io does not provide away to restrict what domains can
```

File: test/unit/providers.test.ts [hunk 2]
```diff
@@ -283,6 +284,19 @@ describe('Providers', () => {
...
+    }
+
+    for (const image of images) {
+      const [, modifiers] = image.args
+      const generated = prepr.getImage('image-test-300x450-png', { modifiers: { ...modifiers }, ...providerOptions }, emptyContext)
+      expect(generated).toMatchObject(image.prepr)
+    }
+  })
+
+
```
Comment (danielroe) [inline test/unit/providers.test.ts:299]:
```suggestion

```

File: test/e2e/__snapshots__/ssr.test.ts.snap [hunk 3]
```diff
@@ -242,6 +242,32 @@ exports[`browser (ssr: true) > none should render images 2`] = `
 ]
 `;
 
+exports[`browser (ssr: true) > prepr should render images 1`] = `
+[
+  "https://nuxt-prepr-demo.stream.prepr.io/q_5//4rhkcc7xzk7d-claymemoirscom-spider-pic.jpg",
```
Comment (danielroe) [inline test/e2e/__snapshots__/ssr.test.ts.snap:247]: should this have two forward slashes in the middle? maybe this should be normalised?

File: test/e2e/__snapshots__/no-ssr.test.ts.snap [hunk 4]
```diff
@@ -132,6 +132,12 @@ exports[`browser (ssr: false) > gumlet should render images 2`] = `
 ]
 `;
 
+exports[`browser (ssr: false) > gumlet should render images 3`] = `
+[
+  "https://demo.gumlet.io/sea.jpeg?w=300&h=300&fit=crop",
+]
+`;
+
```
Comment (danielroe) [inline test/e2e/__snapshots__/no-ssr.test.ts.snap:140]:
```suggestion
```

File: src/runtime/providers/prepr/getImage.ts [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+
+  const projectNameTrimmed = projectName.trim()
+
+  if (projectNameTrimmed === '') {
+    throw new Error('It appears your nuxt configuration includes an empty string at \'image.prepr.projectName\'')
+  }
+
+  const fileBucket = 'stream'
+  const fileOperations = operationsGenerator(options.modifiers)
+  const filePath = fileOperations ? `${fileOperations}/${src}` : src
```
Comment (danielroe) [inline src/runtime/providers/prepr/getImage.ts:39]: possibly could use `joinURL` here?

Comment (danielroe) [reply src/runtime/providers/prepr/getImage.ts:39]:
```suggestion
  const filePath = fileOperations ? joinURL(fileOperations, src) : src
```

File: src/runtime/providers/prepr/getImage.ts [hunk 6]
```diff
@@ -0,0 +1,48 @@
...
+
+  if (typeof projectName !== 'string') {
+    throw new TypeError('The ImageProvider \'prepr\' expects a string to be defined in your nuxt configuration at \'image.prepr.projectName\'')
+  }
+
+  const projectNameTrimmed = projectName.trim()
+
+  if (projectNameTrimmed === '') {
+    throw new Error('It appears your nuxt configuration includes an empty string at \'image.prepr.projectName\'')
+  }
```
Comment (danielroe) [inline src/runtime/providers/prepr/getImage.ts:35]: As this is included in your runtime code, perhaps we could simplify?

```suggestion
  if (typeof projectName !== 'string' || !projectName.trim()) {
    throw new TypeError('[nuxt] [image] [prepr] No project name provided.')
  }
```

File: docs/content/5.providers/prepr.md [hunk 7]
```diff
@@ -0,0 +1,61 @@
...
+to Prepr Support team, more modifiers will be added soon, and the documentation
+page is due for an update.
+
+For the time being you might find the following links useful:
+
+- [Assets Resizing via REST API](https://docs.prepr.io/reference/rest/v1/assets-resizing/)
+- [Understanding your marketing and design team workflows](https://docs.prepr.io/managing-content/images)
+
+::alert{type="info"}
+prepr.io does not provide away to restrict what domains can
```
Comment (danielroe) [inline docs/content/5.providers/prepr.md:34]: oops, missed this:

```suggestion
prepr.io does not provide a way to restrict what domains can
```

File: docs/content/5.providers/prepr.md [hunk 8]
```diff
@@ -0,0 +1,61 @@
...
+})
+```
+
+## Modifiers
+
+The Prepr provider supports a number of additional modifiers. For a full list,
+check out [the Prepr documentation](https://docs.prepr.io/reference/rest/v1/assets-resizing/).
+All current transformations currently mentioned in Prepr docs are supported. According
+to Prepr Support team, more modifiers will be added soon, and the documentation
+page is due for an update.
```
Comment (danielroe) [inline docs/content/5.providers/prepr.md:26]:
```suggestion
All current transformations currently mentioned in Prepr docs are supported.
```

File: test/e2e/__snapshots__/ssr.test.ts.snap (hunk 3 above)
Comment (FranciscoKloganB) [reply test/e2e/__snapshots__/ssr.test.ts.snap:247]: I will test your suggestion see if it addresses this issue. :+1:

File: src/runtime/providers/prepr/getImage.ts (hunk 5 above)
Comment (FranciscoKloganB) [reply src/runtime/providers/prepr/getImage.ts]: Changed in 94b50b66ad36af9ac8f133f6af89a589ede73c77

File: src/runtime/providers/prepr/getImage.ts (hunk 6 above)
Comment (FranciscoKloganB) [reply src/runtime/providers/prepr/getImage.ts]: I have changed to use your suggestion, works either way. I ended up needing to re-`trim()` projectName, later. What do you think?

**Review by danielroe — APPROVED** [review]
I think probably we could inline all the small helper files in the provider directory in a single file, but I'm not really sure it makes much difference.

Thank you for this - it'll be a great addition to the module!

**Pull request merged**